packages/history-utility/src/history-utility.ts

```diff
@@ -101,12 +115,14 @@ export function proxyWithHistory<V>(initialValue: V, skipSubscribe = false) {
      * an object holding the history of snapshots and other metadata <br>
      *   - history.index: the history index to the current snapshot <br>
      *   - history.nodes: the nodes of the history for each change <br>
-     *   - history.wip: field for holding sandbox changes; used to avoid infinite loops<br>
+     *   - history.wip: field for holding sandbox changes; used to avoid infinite loops <br>
+     *   - history.unsubscribe: a function to stop the internal subscription process <br>
      */
     history: ref<History<V>>({
       wip: undefined, // to avoid infinite loop
       nodes: [],
       index: -1,
+      unsubscribe: () => {},
```

Comment:
I'm not sure if having unsubscribe here is a reasonable abstraction. I'd prefer paring `subscribe` and `unsubscribe`. I don't have a good idea, because it subscribe behind the scene, but if we have `unsubscribe` here, we would feel like supporting more edge cases like subscribing twice, unsubscribing twice and so on.

Reply:
i actually tried this but i would have to wrap unsubscribe with ref as well which didnt work out for me

Reply:
this is because it is assigned after subscribe and artificially changes the history nodes

Reply:
I would prefer either of these patterns:
```js
const unsubscribe = foo.subscribe(...);
const unsubscribe = subscribe(foo, ...);
subscribe(foo, ...); unsubscribe(foo, ...);
```

in our case, the last one would be possible. but at the same time, it wouldn't look very nice.

back up a little bit, what's the motivation of unsubscribe in this case, and what are the use cases?

Reply:
I mainly added the unsubscribe method as a way to do cleanup for eg. if a component in react unmounts an d the subscription is no longer needed.

if this is not needed then i guess we can remove it for now.

if it is needed then maybe we move both subscribe and unsubscribe into `history`

Reply:
One solution I could think of _without_ changing the code is:
- note clarify that the default behavior is not able to clean up.
- if one needs to unsubscribe, they should use `skipSubscribe=true` and manually `subscribe()`.

Reply:
> i actually tried this but i would have to wrap unsubscribe with ref as well which didnt work out for me

I think you can define `unsubscribe` method, and internally call a function in the history ref. That seems acceptable.

Reply:
K I will stash this change then and split the pr

Reply:
> if one needs to unsubscribe, they should use skipSubscribe=true and manually subscribe().

Will go with this for now.

in the next PR, I however, extracted the utility to check if the history save should happen. this should reduce the amount of functionality to copy/rewrite for the consumer